src/dotnet/CarbonAware.Plugins.JsonReaderPlugin/CarbonAwareJsonReaderPlugin.cs

```diff
@@ -28,18 +28,16 @@ public async Task<IEnumerable<EmissionsData>> GetEmissionsDataAsync(IDictionary
         List<String> locations = l !=null ? l.ToList() : new List<string>();
 
         var s = props[CarbonAwareConstants.START];
-        DateTime start = s != null ? (DateTime)s : DateTime.Now;
+        DateTime start = s != null ? DateTime.Parse(s.ToString()) : DateTime.Now;
         
         var e = props[CarbonAwareConstants.END];
-        DateTime? end =  e != null ? (DateTime)e : null;
+        DateTime? end =  e != null ? DateTime.Parse(e.ToString()) : null;
 
         var d = props[CarbonAwareConstants.DURATION];
         int durationMinutes =  d!= null ? (int)d : 0;
         
-        foreach (var location in locations)
-        {
-            data = data.Where(ed => ed.Location.Equals(location));        
-
+        if(locations.Any()) {
```

Comment:
nit: space after `if` and `(`
Also make sure that the convention is followed for `{`. Those are below and not at the same line as the statement.
I,e 
```csharp
if (xxx)
{
  ....
}
```